### PR TITLE
Don't unpack WGDOS packed data on FF update

### PR DESCRIPTION
**Pull request by dkillick**
This is a first stab at not unpacking WGDOS packed data when an FF loaded using `iris.experimental.um.FieldsFileVariant` in update mode has changes written to disk.

This is not a completed work! However it does get my progress on this enhancement on here for someone else to take on and finish. The status of the ticket is:
 * the functionality is in place (but see the next point for the caveat to this),
 * changes cannot currently be written to disk by `FieldsFileVariant.close()` due to a type discrepancy (expected `ndarray`, got `str`) when the data is packed. The try-except block in L704-707 demonstrates a start at trying to overcome this.
 * There are no tests for these changes.

### CODE REVIEW

**Comment by pelson** [comment]
Nice approach (and thanks for the guidance @rhattersley). I'm not 100% certain that we shouldn't also be storing other information (such as bpacc or whatever it is called), but as an approach, this certainly seems to be quite clean. I'll *try* to take a look at adopting this today.

File: lib/iris/experimental/um.py [hunk 1]
```diff
@@ -333,12 +346,20 @@ class _NormalDataProvider(object):
...
+        # The lbpack of the field when a instance of this class is created.
+        self.lbpack = lbpack
+
+    def _packed_data_bytes(self, lbnrec):
+        """Return the packed data bytes from disk."""
+        data_size = ((lbnrec * 2) - 1) * _WGDOS_SIZE
+        return self.source.read(data_size)
 
     def read_data(self, field):
+        """Return a NumPy array of the data."""
```
Comment (pelson) [inline lib/iris/experimental/um.py]: This isn't true? It could be bytes I think.

**Comment by pp-mo** [comment]
Supposedly superceded by the reworked version in https://github.com/SciTools/iris/pull/1639
Can we close this now @dkillick ?

**Comment by dkillick** [comment]
Agreed, thanks @pp-mo

**Pull request closed without merging**